auth/store_test.go

```diff
@@ -448,6 +448,26 @@ func TestAuthInfoFromCtx(t *testing.T) {
 	}
 }
 
+func TestAuthDisable(t *testing.T) {
+	as, tearDown := setupAuthStore(t)
+	defer tearDown(t)
+
+	as.AuthDisable()
+
+	ctx := context.WithValue(context.WithValue(context.TODO(), "index", uint64(2)), "simpleToken", "dummy")
+	_, err := as.Authenticate(ctx, "foo", "bar")
+	if err != ErrAuthNotEnabled {
+		t.Errorf("expected %v, got %v", ErrAuthNotEnabled, err)
+	}
+
+	// Disabling disabled auth again and trying to Authenticate
```

Comment:
@Rushit Yea. Probably add the comment into the code as a comment? I know why we need to test it. But it just reads strange without an explanation.